pallets/extrinsic-filter/src/lib.rs

```diff
@@ -0,0 +1,282 @@
+			// just set to the desired mode
+			Mode::<T>::put(mode);
+
+			Self::deposit_event(Event::ModeSet { new_mode: mode });
+
+			Ok(().into())
+		}
+
+		/// block the given extrinsics
+		/// (pallet_name_bytes, function_name_bytes) can uniquely identify an extrinsic
+		/// if function_name_bytes is None, all extrinsics in `pallet_name_bytes` will be blocked
+		#[pallet::weight(10_000)]
+		#[transactional]
+		pub fn block_extrinsics(
+			origin: OriginFor<T>,
+			pallet_name_bytes: Vec<u8>,
+			function_name_bytes: Option<Vec<u8>>,
+		) -> DispatchResultWithPostInfo {
+			T::UpdateOrigin::ensure_origin(origin)?;
+			let pallet_name_string = sp_std::str::from_utf8(&pallet_name_bytes)
+				.map_err(|_| Error::<T>::CannotConvertToString)?;
+			// we disallow blocking this pallet itself
+			ensure!(
+				pallet_name_string != <Self as PalletInfoAccess>::name(),
+				Error::<T>::CannotBlock
+			);
+
+			BlockedExtrinsics::<T>::mutate(
+				(pallet_name_bytes.clone(), function_name_bytes.clone().unwrap_or_default()),
+				|v| {
+					if v.is_none() {
+						*v = Some(());
+						Self::deposit_event(Event::ExtrinsicsBlocked {
+							pallet_name_bytes,
+							function_name_bytes,
+						});
+						// do not pay the fee upon successful block
+						Ok(Pays::No.into())
+					} else {
+						Err(Error::<T>::ExtrinsicAlreadyBlocked.into())
+					}
+				},
+			)
+		}
+
+		/// unblock the given extrinsics
+		/// (pallet_name_bytes, function_name_bytes) can uniquely identify an extrinsic
+		/// if function_name_bytes is None, all extrinsics in `pallet_name_bytes` will be unblocked
+		#[pallet::weight(10_000)]
+		#[transactional]
+		pub fn unblock_extrinsics(
+			origin: OriginFor<T>,
+			pallet_name_bytes: Vec<u8>,
+			function_name_bytes: Option<Vec<u8>>,
+		) -> DispatchResultWithPostInfo {
+			T::UpdateOrigin::ensure_origin(origin)?;
+			if BlockedExtrinsics::<T>::take((
+				pallet_name_bytes.clone(),
+				function_name_bytes.clone().unwrap_or_default(),
+			))
+			.is_some()
+			{
+				Self::deposit_event(Event::ExtrinsicsUnblocked {
+					pallet_name_bytes,
+					function_name_bytes,
+				});
+			} else {
+				return Err(Error::<T>::ExtrinsicNotBlocked.into())
+			}
+			// do not pay the fee upon successful unblock
+			Ok(Pays::No.into())
+		}
+	}
+
+	impl<T: Config> Contains<T::Call> for Pallet<T>
+	where
+		<T as frame_system::Config>::Call: GetCallMetadata,
+	{
+		fn contains(call: &T::Call) -> bool {
+			let allowed_by_mode = match Self::mode() {
+				OperationalMode::Normal => T::NormalModeFilter::contains(call),
+				OperationalMode::Safe => T::SafeModeFilter::contains(call),
+				OperationalMode::Test => T::TestModeFilter::contains(call),
+			};
+			// if allowed_by_mode is true, continue to check BlockedExtrinsics
+			// remember we only have blacklisting, no whitelisting
+			if allowed_by_mode {
+				let CallMetadata { function_name, pallet_name } = call.get_call_metadata();
+				return !BlockedExtrinsics::<T>::contains_key((
```

Comment:
The logic is a bit complex but basically I think it's correct. 
A minor thing, might be picky, but better to swap these two conditions, as it will be false fast in the case function name is default value.

Reply:
> The logic is a bit complex but basically I think it's correct. A minor thing, might be picky, but better to swap these two conditions, as it will be false fast in the case function name is default value.

Thank you.
Not sure if any misunderstanding stands here, but I think that's equally fast? Because it's possible to have both `((palletA, fnA), ())` and `((palletA, ()), ())` at the same time, we have to check both conditions anyway as long as neither is false, which is actually unpredictable (you don't know what was blocked).